scripts/fundTokenLocks.js

```diff
@@ -0,0 +1,120 @@
+/* ./scripts/deployTokenLock.js */
+import { transfers } from './utils/transfers';
+require('log-timestamp');
+const hre = require('hardhat');
+const toBN = hre.ethers.BigNumber.from;
+const fs = require('fs');
+
+const Q18 = toBN(10).pow(toBN(18));
+
+// TODO: move configuration elements fundTokenLocks and deployTockenLocks have in common to utils
+
+// Load the deployment configuration file and set up constants and contract arguments
+const network = hre.hardhatArguments.network;
+console.log('Operating on network: ' + network);
+const actionsFilepath = './scripts/' + network + '/logs/actions.json';
+
+// Load actions object (TODO: move actions function to utils)
+let actions;
+try {
+  // load actions file
+  actions = JSON.parse(fs.readFileSync(actionsFilepath, 'utf-8'));
+} catch (err) {
+  if (err.code === 'ENOENT') {
+    console.log('TokenLock deployment requires an existing actions.json file to retrieve WFAIR contract');
+    process.exit(1);
+  } else {
+    console.error(err);
+    process.exit(1);
+  }
+};
+console.log('Actions will be logged to ' + actionsFilepath);
+// Retrieve WFAIR contract address
+const WFAIR_CONTRACT = actions.token.address;
+// Add transfers key if it doesn't exist
+if (!('transfers' in actions)) { actions.transfers = []; };
+console.log('Number of TokenLock contracts to fund: ' + actions.locks.length);
+console.log(actions.locks);
+
+// Minimum ETH balance - to be determined from gas analysis
+const MIN_ETH = toBN('100000000000000000'); // TODO: should this be in the deployTokenLockConfig.json file?
+
+// Calculate total WFAIR supply requirement, check for cliff/initial conflicts,
+// and create arguments for TokenLock contract
+let TOTAL = toBN(0);
```

Comment:
now this is repeated later in the code. you should also do the following here
1. obtain the Lock address and check the token balance, transfer only the missing amount to what's required
2. you can use `totalLockedTokens` on TokenLock to get the TOTAL directly from the contract. no need to loop :)
3. locks that are in the funded state should be skipped altogether

from this you should obtain of list of transfers to use later in main()

Reply:
as a result this script can be called many time. on the second call (providing the config didn't change) it should have nothing to do